redis/src/caching/mod.rs

```diff
@@ -0,0 +1,18 @@
+//! This module provides **experimental** support for Server-assisted client-side caching in Redis.
+//!
+//! This implementation of Server-assisted client-side cache, only works with RESP3 connections.
+//! Some Redis providers might not enable this feature (e.g. Google Cloud Memorystore).
```

Comment:
lt's not name `Memorystore` here, since this will require us to update the docs if they change their behavior.